ouroboros-network/framework/lib/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -1463,20 +1464,30 @@ with args@Arguments {
                 :: MutableConnState peerAddr handle handleError
                                     version m)
                 <- State.newMutableConnState peerAddr connStateIdSupply connState'
-              -- TODO: label `connVar` using 'ConnectionId'
-              labelTVar connVar ("conn-state-" ++ show peerAddr)
-
-              writeTMVar stateVar
-                        (State.insertUnknownLocalAddr peerAddr mutableConnState state)
-              return ( Just (Left (TransitionTrace
-                                    connStateId
-                                    Transition {
-                                        fromState = Unknown,
-                                        toState   = Known connState'
-                                      }))
-                     , mutableConnState
-                     , Right Nowhere
-                     )
+
+              -- Only proceed if creating a new connection is allowed
+              if inboundRequired connectionMode
+              then do
+                return ( Just (Right (TrInboundConnectionNotFound connectionMode peerAddr))
+                   , mutableConnState
+                   , Left (withCallStack
+                            (InboundConnectionNotFound connectionMode peerAddr))
```

Comment:
I don't think we should error in this case.  It is not an exceptional situation if the connection manager is asked for an `Inbound` connection that is expected to be there at some point.  It would be better to let it return with some particular value to let the caller signal that the connection was not established rather than using exceptions for it.

We'll need a different kind of `Wedge` that has two empty constructors, something isomorphic to `Either (Wedge a b) ()`.